server/pulp/plugins/file/model_distributor.py

```diff
@@ -0,0 +1,249 @@
+        :return:                report describing the publish operation
+        :rtype:                 pulp.plugins.model.PublishReport
+        """
+        progress_report = FilePublishProgressReport(publish_conduit)
+        _logger.info(_('Beginning publish for repository <%(repo)s>') % {'repo': repo.id})
+
+        try:
+            progress_report.state = progress_report.STATE_IN_PROGRESS
+            repo_model = model.Repository.objects.get_repo_or_missing_resource(repo.id)
+            units = repo_controller.find_repo_content_units(repo_model, yield_content_unit=True)
+
+            # Set up an empty build_dir
+            working_dir = common_utils.get_working_directory()
+            build_dir = os.path.join(working_dir, BUILD_DIRNAME)
+            os.makedirs(build_dir)
+
+            self.initialize_metadata(build_dir)
+
+            try:
+                # process each unit
+                for unit in units:
+                    links_to_create = self.get_paths_for_unit(unit)
+                    self._symlink_unit(build_dir, unit, links_to_create)
+                    self.publish_metadata_for_unit(unit)
+            finally:
+                # Finalize the processing
+                self.finalize_metadata()
+
+            # Let's unpublish, and then republish
+            self.unpublish_repo(repo_model, config)
+
+            hosting_locations = self.get_hosting_locations(repo_model, config)
+            for location in hosting_locations:
+                shutil.copytree(build_dir, location, symlinks=True)
+
+            self.post_repo_publish(repo_model, config)
+
+            # Clean up our build_dir
+            common_utils.delete_working_directory()
+
+            # Report that we are done
+            progress_report.state = progress_report.STATE_COMPLETE
+            return progress_report.build_final_report()
+        except Exception, e:
+            _logger.exception(e)
+            # Something failed. Let's put an error message on the report
+            progress_report.error_message = str(e)
+            progress_report.traceback = traceback.format_exc()
+            progress_report.state = progress_report.STATE_FAILED
+            report = progress_report.build_final_report()
+            return report
+
+    def unpublish_repo(self, repo, config):
+        """
+        Delete the published files from our filesystem
+
+        Please also see the superclass method definition for more documentation on this method.
+
+        :param repo: Repository model object
+        :type  repo: pulp.server.db.model.Repository
+
+        :param config: plugin configuration
+        :type  config: pulp.plugins.config.PluginCallConfiguration
+        """
+        hosting_locations = self.get_hosting_locations(repo, config)
+        for location in hosting_locations:
+            self._rmtree_if_exists(location)
+
+    def validate_config(self, repo, config, config_conduit):
```

Comment:
Also not something you created, but I wouldn't know how to implement this since it has no docstring.